mongoengine/base/document.py

```diff
@@ -746,14 +735,15 @@ def merge_index_specs(index_specs, indices):
             if not indices:
                 return index_specs
 
-            spec_fields = [v['fields']
-                           for k, v in enumerate(index_specs)]
+            spec_fields = [idx['fields'] for idx in index_specs]
```

Comment:
Looking at https://docs.mongodb.com/manual/indexes/#create-an-index, I think we're fine. I don't see a valid indexing strategy that would produce a list of lists in there. The only risk is in a mongoengine user defining an invalid index in the code and getting an enigmatic exception about unhashable types. That's not ideal, but should also be relatively easy to figure out.

Given the increased readability of the updated code, I'm tempted to keep it unless you feel strongly about it.